### ISSUE TITLE
bug in SCT checker

### ISSUE THREAD

**Issue by bgregoir** [jasmin-lang/jasmin#724]
The sct checker should reject this program, the return type should be #[public, secret] 
`````
fn xof_init(
  #public reg u64 j,
  #[public, transient] reg ptr u64[25] state,
  #[public, secret] reg ptr u8[32] rho)
  ->
  #[public, transient] reg ptr u64[25]
{
  reg u64 t;
  t = rho[u64 j];
  state[j] = t;
  return state;
}
`````

**Comment by vbgl** [comment]
Beware that annotations are unordered: `#[public, secret]` is the same as `#[secret, public]`.

**Comment by vbgl** [comment]
Suggestion:
~~~
#[constraints="p <= public, t <= transient, secret <= s"]
fn xof_init(
  #public reg u64 j,
  #[poly = p, poly = t] reg ptr u64[25] state,
  #[poly = p, poly = s] reg ptr u8[32] rho)
  ->
  #[poly = p, poly = t] reg ptr u64[25]
{
  reg u64 t;
  t = rho[u64 j];
  state[j] = t;
  return state;
}
~~~

**Comment by bgregoir** [comment]
Why this fix the order ?

**Comment by vbgl** [comment]
Because the two `poly` attributes have the same name (so they stay sorted relative one to each other).

Let’s have a decent way of annotating signatures for security before the next major release.

**Comment by bgregoir** [comment]
Ok, it will do that. Thank a lot.

**Comment by vbgl** [comment]
Fixed by #723.
